app/src/main/java/com/fmsys/snapdrop/JavaScriptInterface.java

```diff
@@ -230,18 +230,67 @@ public static String getSendTextDialogWithPreInsertedString(final String text) {
     }
     public static String initialiseWebsite() {
         return "javascript: " +
+                //change ReceiveTextDialog._onCopy to connect to JavaScriptInterface
+                "ReceiveTextDialog.prototype._oC = ReceiveTextDialog.prototype._onCopy;" +
+                "ReceiveTextDialog.prototype._onCopy = function(){" +
+                "               this._oC();" +
```

Comment:
i think an Event is fired as well, in my tests it works just fine, can you provide a Stacktrace?

Reply:
> The js maybe throws an expection because navigator isnt accessible from inside the WebView, but this should be a silent one; filtering by chromium in Logcat i get Uncaught (in promise) NotAllowedError: Write permission denied. but everything works :)

OK, copy works indeed. Just the optical feedback (Snackbar "Copied to clipboard") is missing. 

> I think we should touch the js as little as possible (add code instead of overriding) not to break anything should there be minor changes to the website

Generally agree. But I think that in case of `_onCopy()` it makes sense to overwrite the code (because we don't want to copy twice ;-) I think there's not much to be expected which could get broken in the future.)

Reply:
I think navigator doesn't work, in webview, so it isn't copied twice.
The snackbar I don't know, but you can add it in the JSI :)